### DISCUSSION TITLE
How to deploy React App on Firebase?

### DISCUSSION THREAD

**Discussion by Bibi1205** [community/community#20496] [Other Feature Feedback, Questions, & Ideas] [answered]
How to deploy React App on Firebase?

**Comment by devcrafta** [comment, marked as answer]
**1.Create react app.**

Prerequisite: node.js, npm, react.js [(tutorial to install)](https://medium.com/nerd-for-tech/how-to-create-and-run-react-js-app-on-macos-a92115f9ee31)

Once we verify that the latest node.js, npm, react.js is installed using following commands.

```
node -v
npm -v
```
Create a react app with your own preffered name using the following command to create a react app.

`npx create-react-app hello-world`
So, once you run the above command, the hello-world directory will be created on your local machine. You can find all of the files you need for your project if you look under the hello-world directory. Simply type the following command, and your simple react application will appear on localhost:3000. npm create-react-app is a command followed by the name of app in my case it is hello-world.

**2. Setup firebase account.**

We’re going to set up Firebase now. To begin, browse to https://firebase.google.com/ and log in to the Firebase Console. You can log in with your Google account and click the create a new project button next to the name of your project. Then follow the on-screen instructions to finish your project.

In firebase console create a project with hello-world
Create a new project and complete the process.

**3. Deploy on firebase**

Open terminal or command propt and navigate to the folder where the project is created …/hello-world.

in the project folder install firebase using following command

`npm install firebase
`Or we can also install firebase globally using following command

`npm install -g firebase-tools
`In your terminal, we’ll need to log into Firebase. If you are not already logged in, you will be asked to provide your login information (email and password). Also, double-check that you’re inside the hello-world folder React application.

`firebase login `

After this you will be a question y/n once you enter y it will take you to browser to login into gmail account and verify your account.

Before initializing firebase.

`npm run build`
Which will create your app production level.

Now we’ll use the following command to set up our Firebase account with our react app:

`firebase init`

Choose option Hosting: for the first question. Set up and launch Firebase Hosting sites.
Select Use an existing project in the second step, Project setup, and then select the Firebase project name that we have created. Our project name is hello-world, so we’ll choose that.
Finally, you’ll need to select the location where Firebase will look for assets to deploy in the Hosting setup section. The production assets will be stored in the build folder by default. As a result, enter build as a response to this option.
Enter y for this option in the Configure as a single-page app question.
Last but not least, you must decide whether or not to overwrite your existing build/index.html file. So select N (No) for this option since we want the actual index.html file generated by Reacts while building the project.
Once the startup is complete, go to the directory and look for two new files. firebase.json, firebaserc These files are created automatically.
Finally deploy your app on firebase using following command.

`firebase deploy
`After you’ve done this command, Firebase will provide you with a unique hosting URL for your deployed app.

Done ! 🎉

**4. References**

Beginner tutorial for react. [[link]](https://medium.com/nerd-for-tech/your-first-react-project-basic-step-by-step-guide-with-examples-and-explinations-d2ed69e6b3c8)
Get started with Firebase Hosting. [[link]](http://get%20started%20with%20firebase%20hosting/)
Create a new react app. [[link]](https://reactjs.org/docs/create-a-new-react-app.html)

**Reply by Bibi1205** [reply]
Dear @Huniko519.
Thank you for your efforts...

**Comment by ws-jm** [comment]
hello.
nice to meet you.

**Comment by ws-jm** [comment]
When developing any project we must host it somewhere so that the whole world can see our hard-work. Hosting websites can be hectic sometimes, but you don’t need to worry as we can now host our React project on Firebase within a minute or two with a very simple setup. 
Creating React Application:
Step 1: Create a React application using the following command:
npx create-react-app react_hosting
Step 2: Install Firebase Tools:
npm install -g firebase-tools
Hosting Website:
Step 1: First, we will initialize a firebase project in our React app by running the following command in the console in our root directory.
Step 2: Select Hosting from the options provided
Step 3: Select the Use an existing project option.
Step 4: As the build folder of react apps is generated in the build folder by default, we will use the same as our public directory.
Step 5: Type y as we want to configure our app as a single-page app.
Step 6: Now we will run the following command to build our react app.
Step 7: Now we just need to run the last command and our application will be deployed.
Output: You can verify that our website was successfully deployed by visiting the hosting link provided by firebase in the console as shown below:

**Comment by fshah21** [comment]
You can follow this guide : https://feyashah.com/how-to-deploy-react-app-using-firebase-hosting
